File: 김현주/점 찍기/점 찍기.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

long long solution(int k, int d) {
    long long answer = 0;
    
    for(int x=0;x<=d;x+=k) {
        long long maxY = sqrt((long long)d*d - (long long)x*x);
        answer += (long long)(maxY/k) + 1;
    }
    
    return answer;
}

File: 김현주/프로그래머스/Lv2/점 찍기/점 찍기.js
function solution(k, d) {
    var answer = 0;    
    // d*d = x*x+ y*y 
    for(let x=0;x<=d;x+=k) {
        let maxY = Math.sqrt(d*d - x*x);
        answer += Math.floor(maxY/k) + 1;
    }
    
    return answer;
}